.github/workflows/cmake.yml

```diff
@@ -0,0 +1,91 @@
+name: CMake Build Matrix
+
+on:
+  push:
+    branches: [ master ]
+  pull_request:
+    branches: [ master ]
+
+env:
+  BUILD_TYPE: Debug
+
+jobs:
+  build:
+    name: ${{ matrix.config.name }}
+    runs-on: ${{ matrix.config.os }}
+
+    strategy:
+      fail-fast: true
+      matrix:
+        config:
+          - {
+            name: "Ubuntu Latest GCC",
+            os: ubuntu-latest,
+            cc: "/usr/bin/gcc",
+            cxx: "/usr/bin/g++",
+            format: "/usr/bin/clang-format-8",
+            tidy: "/usr/bin/clang-tidy-8"
+          }
+          - {
+            name: "Ubuntu Latest Clang",
+            os: ubuntu-latest,
+            cc: "/usr/bin/clang-8",
+            cxx: "/usr/bin/clang++8",
+            format: "/usr/bin/clang-format-8",
+            tidy: "/usr/bin/clang-tidy-8"
+          }
+          - {
+            name: "macOS Latest Clang",
+            os: macos-latest,
+            build_type: "Debug",
```

Comment:
Is there a reason OSX has to redefine build_type?

Reply:
No, I removed it